src/world_model/src/WorldModel.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include "world_model/WorldModel.hpp"
+
+using std::vector;
+using std::string;
+
+WorldModel::WorldModel(const string& name) : Node (name)
+{
+  costmap_ = std::make_unique<Costmap>(this);
+
+  // Create a (lambda) callback function for when a costmap service is received.
```

Comment:
Why a lambda function here versus a class method? Is there an advantage?

Reply:
I'm having problems defining a ROS2 service that takes in a class method for the callback.